### PR TITLE
Tasertron, the Great Dabbening Weapon

### PR DESCRIPTION
**Pull request by SireTurret**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl: Turret, Fucus
add: Added the Tasertron, a makeshift weapon which will be able to fire 4 stun rounds at once by using the power of duct tape and craziness. The downside of such a weapon is that you'll destroy the 3 tasers used in construction, along with the last taser being fully depleted of power.
/:cl:

## About The Pull Request

The tasertron will use 2 power cells, a duct tape, a pipe to wrap the 4 tasers around, 2 DRAGnets to keep them in place and 15 wires to construct it, so it's gonna be hard to make. Additionally, there's only THREE tasers in the armory roundstart, so you'll need to kidnap an officer or wrap your head around it.
## Why It's Good For The Game

a fun little addition to the energy gun family

### CODE REVIEW

File: code/modules/projectiles/guns/energy/stun.dm [hunk 1]
```diff
@@ -46,3 +46,14 @@
...
+/obj/item/gun/energy/e_gun/fivetaser
+	name = "tasertron"
+	desc = "Greytide-issue taser gun, it is able to fire five stun rounds at once but the other five tasers will be destroyed."
+	icon_state = "tasertron"
+	item_state = null //the human update icon will use icon_state instead right
+	ammo_type = list(/obj/item/ammo_casing/energy/electrode)
+	ammo_x_offset = 4
+	can_flashlight = FALSE
+	can_charge = FALSE
+	use_cyborg_cell = FALSE
```
Comment (yoyobatty) [inline code/modules/projectiles/guns/energy/stun.dm]: Doubles?

Comment (SireTurret) [reply code/modules/projectiles/guns/energy/stun.dm]: Hm?

**Comment by SireTurret** [comment]
just realised I should probably put these into the "special" energy gun section

**Comment by yoyobatty** [comment]
You put a copy of the code in tg code and one in hippie code

File: hippiestation/code/modules/crafts/weaponry.dm [hunk 2]
```diff
@@ -18,4 +18,15 @@
 	var/obj/item/shard/new_item = new(user.loc)
 	to_chat(user, "<span class='notice'>You take the duct tape off the [src].</span>")
 	qdel(src)
-	user.put_in_hands(new_item)
\ No newline at end of file
+	user.put_in_hands(new_item)
+
+/obj/item/gun/energy/e_gun/fivetaser
```
Comment (yoyobatty) [inline hippiestation/code/modules/crafts/weaponry.dm]: Here it is again

File: hippiestation/code/modules/projectiles/guns/energy/energy_gun.dm [hunk 3]
```diff
@@ -147,3 +147,10 @@
 				add_overlay("[icon_state]_fail_1")
 			if(151 to INFINITY)
 				add_overlay("[icon_state]_fail_2")
+
+/obj/item/gun/energy/e_gun/fivetaser
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/energy/energy_gun.dm]: And again, the exact same code

**Comment by SireTurret** [comment]
i'll re-do this from scratch since I realised having a different path will just make 3 copies of the same item

**Comment by SireTurret** [comment]
aight removed it from all the useless files

**Comment by SireTurret** [comment]
19 commits oh hell yeah

**Comment by SireTurret** [comment]
for some reason it cant find the path for that crafting

i'll get back to this later

**Comment by SireTurret** [comment]
i've decided to remove the crafting recipe for now until I wrap my head around this shit

**Comment by FantasticFwoosh** [comment]
You can order more tasers by buying premium at cargo with no security access checks as long as you flash the cash, so i don't know how your balancing rule of four in armory and what officers have holds out. (even so, just under armory section ordering regular, you can break the crate open without sec access anyway)

Yep, dragnets easy to import too.

**Comment by SireTurret** [comment]
> You can order more tasers by buying premium at cargo with no security access checks as long as you flash the cash, so i don't know how your balancing rule of four in armory and what officers have holds out. (even so, just under armory section ordering regular, you can break the crate open without sec access anyway)
> 
> Yep, dragnets easy to import too.

since the code I made (not stole) for crafting doesn't even compile I removed it

also, if you complain, please tell me what should be added to the recipe if you think crafting this is too easy

**Comment by SireTurret** [comment]
this should work now

**Comment by SireTurret** [comment]
I'll leave it as an "adminbus" weapon until I add an actual recipe for this.

**Comment by Manly-Dwarf** [comment]
sounds like a gimmick that will never be used

**Comment by SireTurret** [comment]
> sounds like a gimmick that will never be used

for shame

**Comment by Manly-Dwarf** [comment]
let this be lesson for you

**Comment by Manly-Dwarf** [comment]
nah im just kiddin

**Comment by TrickyDickRichard** [comment]
i don't feel like people would use it that much but when they do they will wreck ass

**Comment by FantasticFwoosh** [comment]
> also, if you complain, please tell me what should be added to the recipe if you think crafting this is too easy

I dont think it'll ever be balanced as far as cargonia lock'n'loading because of the problem with the concept rather than the construction; but id suggest it should least take a modular reciever and require a pin put inside to operate as a improvement.

**Comment by Manly-Dwarf** [comment]
i am sorry fwoosh
but firing pins are a garbage feature
they only existed at first to give security players a baby pass since sec players cried about needing to be robust enough to not get disarmed

**Comment by enthethan** [comment]
> i am sorry fwoosh
> but firing pins are a garbage feature
> they only existed at first to give security players a baby pass since sec players cried about needing to be robust enough to not get disarmed

at first, but now they kind of exist  mostly just so science doesn't rush guns since they are useless without raiding the armoury first

**Comment by Maroonenahmoron** [comment]
Something this hilariously bad should make the person shot at go into a seizure or be stunned for an absurd amount of time.

**Review by Manly-Dwarf — APPROVED** [review]
This has a huge problem with it (the icon is stored in the wrong spot) but I'll fix it.

**Pull request merged**